doc/performance-testing/cloud-run/0002_stress-test-phase-2-december-2021.md

```diff
@@ -0,0 +1,131 @@
+| 1  | 116 | 49.5  | 210 | 0.000 | [output](https://console.cloud.google.com/storage/browser/eq-stress-injector-07122021-outputs/stress-test/2021-12-29T08:29:18) |
+| 5  | 488 | 11.5  | 3761 | 0.000 | [output](https://console.cloud.google.com/storage/browser/eq-stress-injector-07122021-outputs/stress-test/2021-12-29T09:14:38) |
+| 10  | 1173  | 35.5  | 157 | 0.000 | [output](https://console.cloud.google.com/storage/browser/eq-stress-injector-07122021-outputs/stress-test/2021-12-29T10:07:10) |
+| 15  |  1755 | 38.5  | 191 | 0.000 | [output](https://console.cloud.google.com/storage/browser/eq-stress-injector-07122021-outputs/stress-test/2021-12-29T11:12:52) |
+| 20  | 2330  |  32.8 | 399 | 0.000 | [output](https://console.cloud.google.com/storage/browser/eq-stress-injector-07122021-outputs/stress-test/2021-12-29T11:49:25) |
+| 25  |  2932 | 40.5  | 148 | 0.000 | [output](https://console.cloud.google.com/storage/browser/eq-stress-injector-07122021-outputs/stress-test/2021-12-29T12:45:40) |
+| 30  |  3507 |  51.5 | 178 | 0.000 | [output](https://console.cloud.google.com/storage/browser/eq-stress-injector-07122021-outputs/stress-test/2021-12-29T13:24:05) |
+| 35  |  4092 | 60.5  | 190 | 0.000 | [output](https://console.cloud.google.com/storage/browser/eq-stress-injector-07122021-outputs/stress-test/2021-12-29T13:59:13) |
+| 40  |  4668 | 64.5  | 229 | 0.000 | [output](https://console.cloud.google.com/storage/browser/eq-stress-injector-07122021-outputs/stress-test/2021-12-29T14:42:36) |
+| 45  | 5228 |  69.5 | 438 | 0.000 | [output](https://console.cloud.google.com/storage/browser/eq-stress-injector-07122021-outputs/stress-test/2021-12-29T15:13:01) |
+| 50  | 1789 | 93.5  | 42127 | 0.240 | [output](https://console.cloud.google.com/storage/browser/eq-stress-injector-07122021-outputs/stress-test/2021-12-29T15:38:08) |
+
+Following completion of the test, re-ran the test with 5 Load Injector Instances as suspected there were issues with the app/load generator, the output was much more in line with the expected pattern:
+
+| Load Injector Instances | Requests per second | 99th percentile Max CPU Usage (%) | 99th percentile response time (ms) | Error rate (%) | Output |
+| --- | --- | --- | --- | --- | --- |
+| 5  | 591 | 10.8  | 155 | 0.000 | [output](https://console.cloud.google.com/storage/browser/eq-stress-injector-07122021-outputs/stress-test/2021-12-31T12:11:15) |
+
+## Errors and Failures
+
+- 30 load injector instances:
+  - 1 x 500 error:
+    `raise common.InvalidResponse("google.resumable_media.common.InvalidResponse: ('Request failed with status code', 503, 'Expected one of', <HTTPStatus.OK: 200>)`
+- 35 load injector instances:
+  - 3 x `Expected a (302) but got a (0) back when posting page`
+- 50 load injector instances:
+  - 1583 x 429 errors:
+      `The request was aborted because there was no available instance.`
+  - 1314  x 500 errors:
+      `The datastore operation timed out, or the data was temporarily unavailable.','grpc_status':4}`
```

Comment:
We need to further understand the cause of the errors. From a brief search of the logs it's not obvious if the GRPC errors had begun prior to the 429 errors. An aggregation of long running Datastore requests would ultimately cause HTTP requests to back up resulting in the Cloud Run 429 (rather than us simply hitting the max instances limit).

Reply:
As discussed, re-ran the the last three tests and added the results to the doc.

Reply:
This is more in line with what I'd expect. I will add some further observations to the write up below following discussions.